CHANGELOG.md

```diff
@@ -8,6 +8,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased]
 
+### Added
+- Support more Python manifest files: `requirements.in`, `setup.py`, `setup.cfg`
+- Recognize all `requirements*.txt` file names as Python lockfiles
+
+### Changed
+- Use `pip-compile` instead of `poetry` to generate lockfiles for `pyproject.toml`
```

Comment:
Isn't it **significantly** more likely that `poetry` is present on a system compared to `pip-compile`?

I saw the tail end of a conversation about this in slack, but I'd still like to bring it up here.

Reply:
Yes, `poetry` is more likely to be present. But `pip-compile` works for any backend, while `poetry` only works for projects using Poetry.

Poetry is certainly the most popular backend, but by using `pip-compile`, we will also support [`flit`](https://github.com/pypa/flit), [`hatch`](https://github.com/pypa/hatch), [`pdm`](https://github.com/pdm-project/pdm), and I'm sure plenty of others that I don't even know about...

Reply:
Yeah but that means you're effectively breaking the most popular backend for the purpose of a couple smaller tools I've never heard of? I'm not sure that's the right move?

Reply:
That's fair. But I don't think we should ignore the smaller tools either. Hatch, in particular, is a PyPA project and probably has the best chance to be distributed with Python in the future.

What if we use a similar approach to yarn/npm here and keep both generators?

Reply:
> What if we use a similar approach to yarn/npm here and keep both generators?

I see no problem with that. I'd personally prefer the `poetry` one if it can do the job and use `pip-tools` for stuff it cannot do.

There's also the option of having a single generator try multiple tools, but I think that's pretty complicated and I'd like to avoid it if possible.

Reply:
Ideally, there would be a direct mapping for each specified `build-backend` in `pyproject.toml` to the tool most closely associate with it (_close_ b/c `build-backend` values can be the same for different `requires` values in the `[build-system]` table; you really need both to figure it out). The issue with using `poetry` (or `pip-compile` or whatever) for all instances of `pyproject.toml` is that the dependency resolvers are different between the tools. The lockfile produced on a user's system should be the same as Phylum produces.

That said, if we pick a single tool to resolve dependencies, `pip-compile` would be my choice since it can handle more types of manifests than just `pyproject.toml`. Plus, limiting the tool count might help when we go to make Docker images for `phylum-ci` to take advantage of the lockfile generation feature.

Reply:
> The issue with using `poetry` (or `pip-compile` or whatever) for all instances of `pyproject.toml` is that the dependency resolvers are different between the tools.

I hadn't thought about this... I was thinking `pip-compile` would use `poetry` when the `pyproject.toml` requires it, but that's only for package metadata and not for resolving the entire dependency tree...

Even more reason to keep both generators (which is what the updated code in this branch does)